cpp/oneapi/dal/backend/primitives/selection/kselect_by_rows_quick.hpp

```diff
@@ -43,10 +45,24 @@ class kselect_by_rows_quick : public kselect_by_rows_base<Float> {
 
 public:
     kselect_by_rows_quick() = delete;
-    kselect_by_rows_quick(sycl::queue& queue, const ndshape<2>& shape)
-            : rnd_seq_(queue, std::min(shape[1], max_rnd_seq_size_)) {
+    kselect_by_rows_quick(sycl::queue& queue, const ndshape<2>& shape) {
         data_ = ndarray<Float, 2>::empty(queue, shape, sycl::usm::alloc::device);
         indices_ = ndarray<std::int32_t, 2>::empty(queue, shape, sycl::usm::alloc::device);
+        rnd_seq_ = ndarray<Float, 1>::empty(queue,
+                                            std::min(shape[1], max_rnd_seq_size_),
+                                            sycl::usm::alloc::shared);
+        auto* rnd_seq_ptr = rnd_seq_.get_mutable_data();
+
+        pr::device_engine engine_gpu = ::oneapi::dal::backend::primitives::device_engine(
+            queue,
+            777,
+            ::oneapi::dal::backend::primitives::engine_type_internal::philox4x32x10);
```

Comment:
Should RNG also be an input parameter in `kselect_by_rows_quick` with `philox4x32x10` as a default value?